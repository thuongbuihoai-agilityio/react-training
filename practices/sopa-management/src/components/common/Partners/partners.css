.partners {
  width: 970px;
  margin: var(--m-5xl) auto;
  padding: var(--px-medium) var(--py-base);
  text-align: center;
  border-radius: var(--rounded-secondary);
  background: var(--bg-primary);
}

.partners-heading {
  width: 780px;
  margin: 0 auto var(--m-4xl);
  font-family: 'var(--font-secondary)';
  font-weight: var(--font-bold);
  line-height: var(--leading-secondary);
}

.partners-image {
  display: flex;
  justify-content: space-between;
}
