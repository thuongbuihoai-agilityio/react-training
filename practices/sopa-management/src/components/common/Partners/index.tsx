// Components
import {
  Icon,
  IconType,
  SizeType,
  Text
} from '@components/common';

// Styles
import './partners.css';

const Partners = () => (
  <div data-testid='partners' className='partners'>
    <Text
      text='“These are the most thoughtfully designed sneakers on the market.”'
      type={SizeType.medium}
      className='partners-heading'
    />
    <div className='partners-image'>
      <Icon iconName={IconType.vogua} />
      <Icon iconName={IconType.esquire} />
      <Icon iconName={IconType.human} />
    </div>
  </div>
);

export default Partners;
