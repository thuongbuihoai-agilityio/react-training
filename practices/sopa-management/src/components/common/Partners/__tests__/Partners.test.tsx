// Libs
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

// Components
import Partners from '@common/Partners';

describe('Partners component', () => {
  test('should render Partners component with type is default', () => {
    const { getByTestId } = render(<Partners />);
    expect(getByTestId('partners')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = render(<Partners />);
    expect(container).toMatchSnapshot();
  });
});
