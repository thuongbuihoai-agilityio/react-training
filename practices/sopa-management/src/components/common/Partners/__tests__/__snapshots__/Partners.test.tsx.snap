// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Partners component matches snapshot 1`] = `
<div>
  <div
    class="partners"
    data-testid="partners"
  >
    <p
      class="partners-heading text-medium"
      data-testid="text"
    >
      “These are the most thoughtfully designed sneakers on the market.”
    </p>
    <div
      class="partners-image"
    >
      <div
        class="undefined icon-vogua"
        data-testid="icon"
      />
      <div
        class="undefined icon-esquire"
        data-testid="icon"
      />
      <div
        class="undefined icon-human"
        data-testid="icon"
      />
    </div>
  </div>
</div>
`;
